plugins/azure-sites-backend/src/service/router.ts

```diff
@@ -52,28 +67,97 @@ export async function createRouter(
     '/:subscription/:resourceGroup/:name/start',
     async (request, response) => {
       const { subscription, resourceGroup, name } = request.params;
-      console.log('starting...');
-      response.json(
-        await azureSitesApi.start({
-          subscription,
-          resourceGroup,
-          name,
-        }),
+      const token = getBearerTokenFromAuthorizationHeader(
+        request.header('authorization'),
       );
+      const entityRef = request.body.entityRef;
+      const entity = await catalogApi.getEntityByRef(entityRef);
```

Comment:
I don't think this will work if authentication is enabled, as you are not passing any token. Make sure to test this

Reply:
Yes, I tested it is working!

Reply:
Hmm, I think this will for sure fail in production without the token. You will need to pass through the `token` from above for all downstream calls.

Reply:
I create a new app and test with Auth enabled. I logged with Azure AD user and did the same test. `Its working!`. Also the `token` (CatalogRequestOptions) is optional in `CatalogClient` for `getEntityByRef`

    getEntityByRef(entityRef: string | CompoundEntityRef, options?: CatalogRequestOptions): Promise<Entity | undefined>;

https://github.com/backstage/backstage/blob/b5ae2e5a62d61f188e1f7b6ac4aeabb8429c6f89/packages/catalog-client/src/CatalogClient.ts#L177-L196

If it intended to work with token only then, I am not sure whether its a bug or not. 

I will pass the `token` in `getEntityByRef` call too.

Reply:
Ah sorry, I mean if you enable `permissions: true` in the backend, then this call will start to fail if there's permissions enabled in the catalog.